packages/headless/src/features/commerce/did-you-mean/did-you-mean-slice.ts

```diff
@@ -0,0 +1,30 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {executeSearch} from '../search/search-actions';
+import {emptyCorrection, getDidYouMeanInitialState} from './did-you-mean-state';
+
+export const didYouMeanReducer = createReducer(
+  getDidYouMeanInitialState(),
+  (builder) => {
+    builder
+      .addCase(executeSearch.pending, (state) => {
```

Comment:
Since the code is very similar to did-you-mean for classic search, we should do what Fred did here: https://github.com/coveo/ui-kit/commit/c5d8f2efc65c3a523f35287d09c8cf56ab1fbd1f#diff-5906e55e79abd245ae48d83ce007602dd735e1d8c20c1148859d870565bf7979R1-R69

Create a `did-you-mean-functions` file, which we reuse for both reducers.